app/frontend/src/features/communities/leaveReference/ReferenceDataContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+export type ReferenceContextFormData = {
+  text: string;
+  wasAppropriate: string;
+  rating: number;
+};
+
+interface ReferenceDataProviderProps {
+  children: ReactNode;
+}
+
+type ReferenceContextType = {
+  data: ReferenceContextFormData;
+  setValues: (data: ReferenceContextFormData) => void;
+};
+
+const ReferenceDataContext = createContext<ReferenceContextType | undefined>(
+  undefined
+);
+
+export const ReferenceDataProvider = ({
+  children,
+}: ReferenceDataProviderProps) => {
+  const [data, setData] = useState({
+    text: "",
+    wasAppropriate: "",
+    rating: 0,
+  });
+
+  const setValues = (values: any) => {
```

Comment:
Better not use `any`. You could do `typeof data`, but why not just provide setData directly?

Reply:
Because the `data` is an array, if we use `setData` we would have to pass a value for every previous field, while now with `setValues` we merge the new values with the previous values.